packages/components/table/src/store/watcher.ts

```diff
@@ -201,15 +209,16 @@ function useWatcher<T>() {
     let deleted
     if (rowKey.value) {
       deleted = []
-      const selectedMap = getKeysMap(selection.value, rowKey.value)
       const dataMap = getKeysMap(data.value, rowKey.value)
-      for (const key in selectedMap) {
-        if (hasOwn(selectedMap, key) && !dataMap[key]) {
-          deleted.push(selectedMap[key].row)
+      for (const key in selectedMap.value) {
+        if (hasOwn(selectedMap.value, key) && !dataMap[key]) {
+          deleted.push(selectedMap.value[key].row)
         }
       }
     } else {
-      deleted = selection.value.filter((item) => !data.value.includes(item))
+      deleted = selection.value.filter((item) => {
+        return !data.value.some((row) => isEqual(row, item))
```

Comment:
Why do we use `isEqual` here? Is it more efficient than includes?

Reply:
If the user changes the `data` value and then changes it back to the previous value (eg. switching pages), the correct value cannot be matched.

Reply:
This is a breaking change, which will be inconsistent with the original performance. We should not put this change in this PR. If we are sure to change it, we can open a separate PR and not mix it with what this PR does.